docs/pages/build-reference/e2e-tests.mdx

```diff
@@ -199,28 +237,81 @@ Edit **eas.json** and add the `test` build profile:
 }
 ```
 
-Create **eas-hooks/eas-build-pre-install.sh** that will be used to install `applesimutils` with `brew`:
+Create **eas-hooks/eas-build-pre-install.sh** that installs necessary tools and dependencies for the given platform:
 
 ```sh eas-hooks/eas-build-pre-install.sh
 #!/usr/bin/env bash
 
 set -eox pipefail
 
-if [[ "$EAS_BUILD_PLATFORM" == "ios" && "$EAS_BUILD_PROFILE" == "test" ]]; then
-  brew tap wix/brew
-  brew install applesimutils
+if [[ "$EAS_BUILD_RUNNER" == "eas-build" && "$EAS_BUILD_PROFILE" == "test"* ]]; then
+  if [[ "$EAS_BUILD_PLATFORM" == "android" ]]; then
+    sudo apt-get --quiet update --yes
+
+    # Install emulator & video bridge dependencies
+    # Source: https://github.com/react-native-community/docker-android/blob/master/Dockerfile
+    sudo apt-get --quiet install --yes \
+      libc6 \
+      libdbus-1-3 \
+      libfontconfig1 \
+      libgcc1 \
+      libpulse0 \
+      libtinfo5 \
+      libx11-6 \
+      libxcb1 \
+      libxdamage1 \
+      libnss3 \
+      libxcomposite1 \
+      libxcursor1 \
+      libxi6 \
+      libxext6 \
+      libxfixes3 \
+      zlib1g \
+      libgl1 \
+      pulseaudio \
+      socat
+
+    sdkmanager --install "system-images;android-32;google_apis;x86_64"
+    avdmanager --verbose create avd --force --name "pixel_4" --device "pixel_4" --package "system-images;android-32;google_apis;x86_64"
+  else
+    brew tap wix/brew
+    brew install applesimutils
+  fi
 fi
+
 ```
 
-Next, create **eas-hooks/eas-build-on-success.sh** with the following contents. The script runs Detox tests in headless mode:
+Next, create **eas-hooks/eas-build-on-success.sh** with the following contents. The script runs different commands for Android and iOS. For iOS, the only command is `detox test`. For Android, it's a bit more complicated. You'll have to start the emulator prior to running the tests as `detox` sometimes seems to be having problems with starting the emulator on its own and it can stuck at running the first test from your test suite. After the `detox test` run, there is a command that kills the previously started emulator.
```

Comment:
```suggestion
Next, create **eas-hooks/eas-build-on-success.sh** with the following contents. The script runs different commands for Android and iOS. For iOS, the only command is `detox test`. For Android, it's a bit more complicated. You'll have to start the emulator prior to running the tests as `detox` sometimes seems to be having problems with starting the emulator on its own and it can get stuck on running the first test from your test suite. After the `detox test` run, there is a command that kills the previously started emulator.
```